[DatePicker] Add option to hide date display

dhoward opened the pull request:
This adds an option to hide the date display above the date picker (or on the side on portrait mode). I did not add any unit tests because this is purely a visual change, but the code has been linted and docs have been added.

Thanks for the awesome library!

- [x] PR has tests / docs demo, and is linted.
- [x] Commit and PR titles begin with [ComponentName], and are in imperative form: "[Component] Fix leaky abstraction".
- [x] Description explains the issue / use-case resolved, and auto-closes the related issue(s) (http://tr.im/vFqem).

<img width="319" alt="screen shot 2017-02-15 at 10 14 30 pm" src="https://cloud.githubusercontent.com/assets/268974/23009723/650d7e34-f3cc-11e6-81e5-d61261a2d8c7.png">

stunaz commented:
By doing so, you lost the ability to choose the years

dhoward commented:
@stunaz yes that is true, sorry I forgot to mention that. I figured it would be ok since you already have a `disableYearSelection` option for users who do not need that functionality. I could add a line in the documentation something like “Hiding the Date Display will also hide year selection” to clarify that this will hide both. Does that sound ok?

oliviertassinari commented on src/DatePicker/Calendar.js:
```diff
@@ -42,6 +42,7 @@ class Calendar extends Component {
     onTouchTapOk: PropTypes.func,
     open: PropTypes.bool,
     shouldDisableDate: PropTypes.func,
+    showDateDisplay: PropTypes.bool,
```
I have some consideration for the naming convention and the API consistency. With boolean properties, we have two options. `showX`/`hideX` I think that we should be choosing based on the default value. That's basically what have been done at the HTML specification, e.g. `disabled`, `readonly` in order to allow the shorthand.
Let's follow the same convention. with `hideX` as the default is going to be `false`.

TL;DR What about `hideCalendarDate`?

mbrookes commented:
> By doing so, you lost the ability to choose the years

#4949 would solve that.

dhoward commented:
@oliviertassinari thanks for the feedback, I have renamed the prop and added unit tests for showing/hiding the date. 

It looks like TravisCI is failing due to lint errors in a couple unrelated files, I see at least one of them is already being addressed in #6182, not sure about the other one but I can add a fix on this PR if that would be helpful.

oliviertassinari commented:
Thanks.
